### PR TITLE
boards: stm32f746g_disco: Enable Ethernet

### PR DESCRIPTION
**Pull request by chrta**
This adds Ethernet support for the ST STM32F746G Discovery board.

Also a minor typo in eth_stm32_hal.c is fixed and some information are added to the board yaml file.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

### CODE REVIEW

**Review by jukkar — COMMENTED** [review]
Looks ok, note that sanitychecker fails as you have a commit without anything in body.

File: boards/arm/stm32f746g_disco/stm32f746g_disco.yaml [hunk 1]
```diff
@@ -5,3 +5,9 @@ arch: arm
 toolchain:
   - zephyr
   - gnuarmemb
+ram: 320
+flash: 1024
+supported:
+  - netif:eth
+  - gpio
+
```
Comment (pfalcon) [inline boards/arm/stm32f746g_disco/stm32f746g_disco.yaml]: Probably this extra line worth fixing.

File: boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst [hunk 2]
```diff
@@ -100,6 +100,8 @@ The Zephyr stm32f746g_disco board configuration supports the following hardware
 +-----------+------------+-------------------------------------+
 | GPIO      | on-chip    | gpio                                |
 +-----------+------------+-------------------------------------+
+| ETHERNET  | on-chip    | Ethernet                            |
```
Comment (erwango) [inline boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst]: Can you update pinmuxing section too?

**Review by erwango — CHANGES_REQUESTED** [review]
One comment otherwise is fine

File: boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst (hunk 2 above)
Comment (chrta) [reply boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst]: Sure. Can you point be to a board doc and the relevant section, so i see what you are expecting?

Comment (erwango) [reply boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst]: cf: https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/nucleo_f429zi/doc/nucleof429zi.rst#default-zephyr-peripheral-mapping

**Review by jukkar — APPROVED** [review]
LGTM

File: boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst (hunk 2 above)
Comment (chrta) [reply boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst]: Should be fixed now

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**